### ISSUE TITLE
Firefox has very high CPU utilization with NoScript even when the browser is idle

### ISSUE THREAD

**Issue by baharev** [hackademix/noscript#111]
Up until yesterday, FF worked as expected; thanks to NoScript, the CPU utilization was always low. 

I did not change anything (or at least not on purpose) but I noticed that FF suddenly saturates 2-3 cores of my 4 core machine even when the browser is idle.

After poking around for a while, it turned out that if I *disable* NoScript at the add-ons, the CPU utilization drops to near zero when the browser is idle. It is 100% reproducible, deterministic. So I am pretty convinced the NoScript is causing it. :-(

I upgraded the browser, now running 

`Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:70.0) Gecko/20100101 Firefox/70.0`

and I have also tried the latest development version of NoScript:

`11.0.6rc3`

Yet, FF consumes 2-3 of the 4 cores. Here is an example:

![Screenshot_2019-11-02_20-46-53](https://user-images.githubusercontent.com/2605584/68076460-16816f00-fdb5-11e9-95e4-0e6052b55f3d.png)

I have 8 pinned tabs. Up until yesterday, when the browser was idle, the CPU utilization was near zero.

**How can I debug this? What piece of information would help you?**

In the meantime, is there a workaround so that I can get FF back? 

Disabling NoScript is not an option for me, nor is saturating 2-3 cores of my machine.

**Comment by ActuallyaDeviloper** [comment]
I am also affected by this issue but on Windows.

I noticed that my CPU fan was was constantly running. Then I tried disabling different add-ons until I found that NoScript was the culprit.

As workaround I have now manually reinstalled an older version of NoScript from the 20th of August from my backups.

**Comment by baharev** [comment]
@ActuallyaDeviloper Indeed, rolling back to 11.0.3 from Aug 19, 2019 and also disabling automatic updates is an acceptable workaround for the time being. Many thanks for the tip!

By the way, you do not need any backup, you can roll back here:

https://addons.mozilla.org/en-US/firefox/addon/noscript/versions/

And as I said, don't forget to turn off automatic updates.

Now we know that the bug was introduced *after* 11.0.3, most likely with 11.0.6 which was released on Nov 1, when my problems started. I hope this helps the developer to narrow down the changes that could have caused the problem.

**Comment by hackademix** [comment]
Please check 11.0.7rc1 from https://noscript.net/getit#devel, thanks.

**Comment by baharev** [comment]
The [noscript website](https://noscript.net/getit) is out of sync compared to [Mozilla's website for the add-on](https://addons.mozilla.org/en-US/firefox/addon/noscript/versions/).

The noscript website says:

> You can get latest stable version here, too, using this direct download link for NoScript 11.0.4

and indeed, 11.0.4 is linked, no typo.

That's why I thought yesterday I was testing the latest development version `11.0.6rc3`. However, on the Mozilla website I see that Version 11.0.6 Released Nov 1, 2019, so even "the latest" development version was out of sync yesterday. Today at least the latest development version seems up to date (11.0.7rc1).

**Comment by baharev** [comment]
@hackademix I have installed 11.0.7rc1 and browsed as usual. The CPU usage seems to be back to normal, and the problem seems to be gone. So apparently you have figured out what caused it and fixed it, but let's wait for feedback from @ActuallyaDeviloper too before closing this issue.

A big thank you for this add-on!

**Comment by ActuallyaDeviloper** [comment]
Thanks, seems to have fixed it for me as well.

**Comment by phil294** [comment]
I am seeing this issue today. Installed for the first time (11.4.4), and 30-40% cpu utilization. Maybe it's because I have 15 windows open right now, but it's surely because of noscript.
